sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -725,6 +728,36 @@ def _check_connections_by_use_db(self):
                 # Switch DB back to MASTER
                 cursor.execute(SWITCH_DB_STATEMENT.format(self.connection.DEFAULT_DATABASE))
 
+    def get_databases(self):
+        engine_edition = self.static_info_cache.get(STATIC_INFO_ENGINE_EDITION)
+        if not is_azure_sql_database(engine_edition):
+            db_names = [d.name for d in self.databases] or [
+                self.instance.get('database', self.connection.DEFAULT_DATABASE)
+            ]
+        else:
+            db_names = [self.instance.get('database', self.connection.DEFAULT_DATABASE)]
+        return db_names
+
+    def do_for_databases(self, action, databases):
+        exceptions = []
+        engine_edition = self.static_info_cache.get(STATIC_INFO_ENGINE_EDITION)
+        with self.connection.open_managed_default_connection():
+            with self.connection.get_managed_cursor() as cursor:
+                for db in databases:
+                    try:
+                        if not is_azure_sql_database(engine_edition):
+                            cursor.execute(SWITCH_DB_STATEMENT.format(db))
+                        action(cursor, db)
+                    except StopIteration:
+                        exceptions.append((db, "StopIteration"))
+                        return exceptions
+                    except Exception as e:
```

Comment:
If an exception is thrown here, I think it makes sense to log it and then `continue` in the loop so we don't stop collecting for the remaining databases in the list